mantidproject/mantid
# ALC with nxs2 and bin files

**Description of work.**

This is to add the ability to read file formats into the ALC, other than nxs. The two examples this has focused on are the nxs_v2 (from ISIS) and bin (PSI) muon data formats. This is needed as ISIS will be moving to primarily using nxs_v2 after the long shut down.

**To test:**


1. Get the files from me
2. Open Mantid and set the mange user directories to have: no archive, set it to search the files provided
3. Set the instrument to HIFI

Enter each of the following run numbers in turn and repeat the following steps:
175958-61 (nxs_v2)
175958-61, 175965 (nxs and nxs_v2 mix: no other mixes should be possible)
1808-13 (PSI bin format, which has been fudged to look like HIFI data by changing the name of the file)
1529 (PSI bin format and extra logs. This should have extra logs compared to the above)

For each set of runs:
1. Check the log values in the drop down
2. Press the load button
3. Check you get a nice plot



Fixes #32123. 



In the release notes as can read nxs_v2 (the bin changes were made to check that Load works correctly and not fully available to users)



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
